pkg/alexa/l10n/l10n.go

```diff
@@ -0,0 +1,246 @@
+package l10n
+
+import (
+	"fmt"
+	"github.com/drpsychick/alexa-go-cloudformation-demo/pkg/alexa"
+	"math/rand"
+	"time"
+)
+
+// Default keys
+const (
+	KeySkillName                Key = "SKILL_Name"
+	KeySkillDescription         Key = "SKILL_Description"
+	KeySkillSummary             Key = "SKILL_Summary"
+	KeySkillExamplePhrases      Key = "SKILL_ExamplePhrases"
+	KeySkillKeywords            Key = "SKILL_Keywords"
+	KeySkillSmallIconURI        Key = "SKILL_SmallIconURI"
+	KeySkillLargeIconURI        Key = "SKILL_LargeIconURI"
+	KeySkillTestingInstructions Key = "SKILL_TestingInstructions"
+	KeySkillInvocation          Key = "SKILL_Invocation"
+	KeySkillPrivacyPolicyURL    Key = "SKILL_PrivacyPolicyURL"
+	KeySkillTermsOfUse          Key = "SKILL_TermsOfUse"
+)
+
+func init() {
+	rand.Seed(time.Now().UTC().UnixNano())
+}
+
+func Speak(text string) string {
+	return "<speak>" + text + "</speak>"
+}
+func UseVoice(voice string, text string) string {
+	return "<voice name=\"" + voice + "\">" + text + "</voice>"
+}
+
+func UseVoiceLang(voice string, language string, text string) string {
+	return "<voice name=\"" + voice + "\"><lang xml:lang=\"" + language + "\">" + text + "</lang></voice>"
+}
+
+// Registry is the Locale registry
+type Registry struct {
+	defaultLocale string
+	locales       map[string]*Locale
+}
+
+// DefaultRegistry is the standard registry used
+var DefaultRegistry = &Registry{
+	locales: map[string]*Locale{},
+}
+
+// Locale is a representation of Keys in a specific language (and can have a fallback Locale)
+type Locale struct {
+	Name            string          // de-DE, en-US, ...
+	Countries       []alexa.Country // countries associated with this locale
+	Invocation      string          // "my skill"
+	Fallback        *Locale         // points to fallback (or nil)
+	TextSnippets    Snippets
+	IntentResponses IntentResponses
+}
+
+// Key defines the type of a text key
+type Key string
+
+func (k Key) String() string {
+	return string(k)
+}
+
+// Snippets is the actual representation of key -> array of texts in locale
+type Snippets map[Key][]string
+
+// Responses is the representation of a list of IntentResponses
+type IntentResponses map[Key]IntentResponse
+
+type IntentResponse struct {
```

Comment:
This seems crazy specific and a little complex. Is it really needed?

Reply:
It makes things easier, for the translator (everything concerning one intent is in one place) and for processing: fetch all translations needed from one key.

I agree that in some cases I chose "easy" over "simple" and may need refactoring to remove complexity.

Reply:
It makes things easier for 1 very specific use case. If i choose to not return translations this way, you have made things really difficult for me to figure out what to do. There are many use cases to not return cards, or to return text instead of SSML, or to return a different type of card. None of these are covered here. You are needless complicating a very simple action with this. Image the idea that in 1 language you use text and another you use SSML. How is this covered?

Reply:
completely removed `IntentResponse`